keanu-project/src/main/proto/io/improbable/keanu/BayesianNetwork.proto

```diff
@@ -7,30 +7,57 @@ option java_outer_classname = "KeanuSavedBayesNet";
 
 message BayesianNetwork {
     repeated Vertex vertices = 1;
-    repeated VertexValue defaultState = 2;
+    repeated StoredValue defaultState = 2;
+}
+
+message VertexID {
+    string id = 1;
 }
 
 message Vertex {
-    repeated int64 id = 1;
+    VertexID id = 1;
     string label = 2;
     string vertexType = 3;
     repeated NamedParent parents = 5;
+    VertexValue constantValue = 6;
+}
+
+message StoredValue {
+    VertexValue value = 1;
+    string vertexLabel = 2;
+    VertexID id = 3;
+}
+
+message DoubleTensor {
+    repeated int64 shape = 1;
+    repeated double values = 2;
+}
+
+message IntegerTensor {
+    repeated int64 shape = 1;
+    repeated int32 values = 2;
+}
+
+message BooleanTensor {
+    repeated int64 shape = 1;
+    repeated bool values = 2;
+}
+
+message GenericTensor {
+    repeated int64 shape = 1;
+    repeated bytes values = 2;
```

Comment:
Are you going to infer the number of bytes in the individual value from the shape?

Reply:
TBH: I needed to discuss this with you - I'm not sure how GenericTensors really work at the moment.  I was going to put their implementation in a separate PR though once we're happy with the overall approach.  Certainly this structure mirrors what is stored inside the generic Tensor itself though.